File: website/src/main/java/info/xiaomo/website/service/impl/WorksServiceImpl.java
package info.xiaomo.website.service.impl;

import info.xiaomo.website.dao.WorksDao;
import info.xiaomo.website.model.WorksModel;
import info.xiaomo.website.service.WorksService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;
import org.springframework.stereotype.Service;

import java.util.Date;
import java.util.List;
import java.util.Optional;

/**
 * 把今天最好的表现当作明天最新的起点．．～
 * いま 最高の表現 として 明日最新の始発．．～
 * Today the best performance  as tomorrow newest starter!
 * Created by IntelliJ IDEA.
 *
 * @author : xiaomo
 * github: https://github.com/xiaomoinfo
 * email: xiaomo@xiaomo.info
 * <p>
 * Date: 2016/11/3 14:34
 * Copyright(©) 2015 by xiaomo.
 **/

@Service
public class WorksServiceImpl implements WorksService {

    private WorksDao dao;

    @Autowired
    public WorksServiceImpl(WorksDao dao) {
        this.dao = dao;
    }

    @Override
    public List<WorksModel> findAll() {
        return dao.findAll();
    }

    @Override
    public Page<WorksModel> findAll(int start, int pageSize) {
        return dao.findAll(PageRequest.of(start - 1, pageSize));
    }

    @Override
    public WorksModel findById(Long id) {
        Optional<WorksModel> optionalModel = dao.findById(id);
        return optionalModel.orElse(null);
    }

    @Override
    public WorksModel findByName(String name) {
        return dao.findWorkByName(name);
    }

    @Override
    public WorksModel update(WorksModel model) {
        Optional<WorksModel> optional = dao.findById(model.getId());
        if (!optional.isPresent()) {
            return null;
        }
        WorksModel result = optional.get();
        if ("".equals(model.getCompleteTime()) && model.getCompleteTime() != null) {
            result.setCompleteTime(model.getCompleteTime());
        }
        if ("".equals(model.getImgUrl()) && model.getImgUrl() != null) {
            result.setImgUrl(model.getImgUrl());
        }
        if ("".equals(model.getName()) && model.getName() != null) {
            result.setName(model.getName());
        }
        if ("".equals(model.getSummary()) && model.getSummary() != null) {
            result.setSummary(model.getSummary());
        }
        if ("".equals(model.getUrl()) && model.getUrl() != null) {
            result.setUrl(model.getUrl());
        }
        result.setUpdateTime(new Date());
        return dao.save(result);
    }

    @Override
    public WorksModel add(WorksModel model) {
        model.setCreateTime(new Date());
        model.setUpdateTime(new Date());
        return dao.save(model);
    }

    @Override
    public void del(Long id) {
        Optional<WorksModel> optional = dao.findById(id);
        if (!optional.isPresent()) {
            return;
        }
        dao.delete(optional.get());
    }
}

File: website/src/main/java/info/xiaomo/website/service/impl/WorksServiceImpl.kt
package info.xiaomo.website.service.impl

import info.xiaomo.website.dao.WorksDao
import info.xiaomo.website.model.WorksModel
import info.xiaomo.website.service.WorksService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.data.domain.Page
import org.springframework.data.domain.PageRequest
import org.springframework.stereotype.Service
import java.util.*

/**
 * 把今天最好的表现当作明天最新的起点．．～
 * いま 最高の表現 として 明日最新の始発．．～
 * Today the best performance  as tomorrow newest starter!
 * Created by IntelliJ IDEA.
 *
 * @author : xiaomo
 * github: https://github.com/xiaomoinfo
 * email: xiaomo@xiaomo.info
 *
 * Date: 2016/11/3 14:34
 * Copyright(©) 2015 by xiaomo.
 */

@Service
class WorksServiceImpl @Autowired
constructor(private val dao: WorksDao) : WorksService {

    override fun findAll(): List<WorksModel> {
        return dao.findAll()
    }

    override fun findAll(start: Int, pageSize: Int): Page<WorksModel> {
        return dao.findAll(PageRequest(start - 1, pageSize))
    }

    override fun findById(id: Long?): WorksModel {
        return dao.findOne(id)
    }

    override fun findByName(name: String): WorksModel {
        return dao.findWorkByName(name)
    }

    override fun update(model: WorksModel): WorksModel? {
        val result = dao.findOne(model.id) ?: return null
        if ("" == model.completeTime && model.completeTime != null) {
            result.completeTime = model.completeTime
        }
        if ("" == model.imgUrl && model.imgUrl != null) {
            result.imgUrl = model.imgUrl
        }
        if ("" == model.name && model.name != null) {
            result.name = model.name
        }
        if ("" == model.summary && model.summary != null) {
            result.summary = model.summary
        }
        if ("" == model.url && model.url != null) {
            result.url = model.url
        }
        result.updateTime = Date()
        return dao.save(result)
    }

    override fun add(model: WorksModel): WorksModel {
        model.createTime = Date()
        model.updateTime = Date()
        return dao.save(model)
    }

    override fun del(id: Long?) {
        dao.delete(id)
    }
}
